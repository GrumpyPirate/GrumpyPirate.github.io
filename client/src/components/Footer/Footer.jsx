import React from 'react';

import Icon from 'components/Icon/Icon';

import './Footer.scss';

const Footer = () => {
  const creationYear = 2017;
  const currentYear = new Date().getFullYear();

  return (
    <footer className="footer text-center">
      <div className="footer__credits">
        <div className="container-fluid">
          <div className="last-child-mb-0">
            <p className="mb-0">
              {'Made with '}
              <a
                href="https://facebook.github.io/react/"
                target="_blank"
                rel="noopener noreferrer"
                className="footer__icon link-inherit"
                title="React"
              >
                <Icon glyph="react" altText="React" />
              </a>
              {', '}
              <a
                href="https://webpack.js.org/"
                target="_blank"
                rel="noopener noreferrer"
                className="footer__icon link-inherit"
                title="Webpack"
              >
                <Icon glyph="webpack" altText="Webpack" />
              </a>
              {', '}
              <a
                href="https://www.contentful.com/"
                target="_blank"
                rel="noopener noreferrer"
                className="footer__icon link-inherit"
                title="Contentful"
              >
                <Icon glyph="contentful" altText="Contentful" />
              </a>
              {' and '}
              <span className="footer__icon">
                <Icon glyph="heart" altText="love" />
              </span>
            </p>

            <p>
              &copy;
              {' '}
              {currentYear > creationYear
                ? <span className="footer__credits__date">{creationYear}&mdash;{currentYear}</span>
                : <span className="footer__credits__date">{currentYear}</span>
              }
              {' '}
              Edward Cobbold
            </p>
          </div>
        </div>
      </div>
    </footer>
  );
};

export default Footer;
