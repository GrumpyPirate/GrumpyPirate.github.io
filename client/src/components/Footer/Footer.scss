@import 'functions';
@import 'variables';
@import 'mixins';

.footer {
  background-color: palette(theme-dark);
  color: mix(palette(theme-light), palette(theme-dark), 40%);
  padding: 1.5rem 0;
  font-size: $small-font-size;

  // md up
  @include media-breakpoint-up(md) {
    padding: 2rem 0;
  }

  &__icon {
    vertical-align: middle;
    font-size: 175%;

    .icon {
      bottom: .125em;
    }
  }
}
