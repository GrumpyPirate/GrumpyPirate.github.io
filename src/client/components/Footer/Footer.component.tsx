import React from 'react';
import classnames from 'classnames';

import Container from 'components/Layout/Container/Container.component';
import Icon from 'components/Icon/Icon.component';

import classes from './Footer.component.scss';

const Footer = () => {
  const creationYear = 2017;
  const currentYear = new Date().getFullYear();

  return (
    <footer className={classes['footer']}>
      <Container>
        <p className={classes['footer__tech']}>
          'Made with '
          <a
            href="https://facebook.github.io/react/"
            target="_blank"
            rel="noopener noreferrer"
            title="React"
            className={classnames(classes['footer__link'], classes['footer__link--icon'])}
          >
            <Icon glyph="react" altText="React" />
          </a>
          ', '
          <a
            href="https://webpack.js.org/"
            target="_blank"
            rel="noopener noreferrer"
            title="Webpack"
            className={classnames(classes['footer__link'], classes['footer__link--icon'])}
          >
            <Icon glyph="webpack" altText="Webpack" />
          </a>
          ' and '
          <a
            href="https://www.contentful.com/"
            target="_blank"
            rel="noopener noreferrer"
            title="Contentful"
            className={classnames(classes['footer__link'], classes['footer__link--icon'])}
          >
            <Icon glyph="contentful" altText="Contentful" />
          </a>
          .{' '}
          <a
            href="https://bitbucket.org/GrumpyPirate/portfolio-2017"
            className={classes['footer__link']}
            target="_blank"
            rel="noreferrer noopener"
          >
            Source code
          </a>
          .
        </p>

        <p className={classes['footer__copyright']}>
          &copy;{' '}
          {currentYear > creationYear ? (
            <span className={classes['footer__copyright__date']}>
              {creationYear}&mdash;{currentYear}
            </span>
          ) : (
            <span className={classes['footer__copyright__date']}>{currentYear}</span>
          )}{' '}
          <a
            className={classes['footer__link']}
            href="https://github.com/grumpypirate"
            target="_blank"
            rel="noopener noreferrer"
          >
            Edward Cobbold
          </a>
          .
        </p>
      </Container>
    </footer>
  );
};

export default Footer;
