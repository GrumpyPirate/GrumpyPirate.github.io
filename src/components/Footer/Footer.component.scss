@import 'core';

.footer {
  background-color: palette('theme-dark-shades');
  color: transparentize(palette('theme-light-shades'), 0.5);
  font-size: rem(12px);
  padding: 1.5rem 0;
  text-align: center;

  // md up
  @include respond-up('md') {
    padding: 2rem 0;
  }

  // Elements
  &__tech {
    margin-bottom: 0;
  }

  &__copyright {
    margin-bottom: 0;
  }

  &__link {
    @include no-underline;

    color: inherit;
    font: inherit;

    &--icon {
      font-size: 1.75em;
      vertical-align: middle;
    }

    &:hover,
    &:focus {
      color: palette('theme-accent-light');
    }
  }
}
