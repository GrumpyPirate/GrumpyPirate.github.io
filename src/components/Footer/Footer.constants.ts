import styled from 'styled-components';

import { noUnderline, palette } from 'styles';

export const Tech = styled.p`
  margin-bottom: 0;
`;

export const Copyright = styled.p`
  margin-bottom: 0;
`;

export const FooterLink = styled.a`
  ${noUnderline}

  color: inherit;
  font: inherit;

  &:hover,
  &:focus {
    color: ${palette.themeAccentLight};
  }
`;

export const FooterLinkWithIcon = styled(FooterLink)`
  font-size: 1.75em;
  vertical-align: middle;
`;
