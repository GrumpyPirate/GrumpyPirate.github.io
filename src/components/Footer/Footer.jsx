// React
import React from 'react'

import Icon from 'components/Icons/Icons.jsx'

// SCSS
import './Footer.scss'

const Footer = () => {
  const CREATION_YEAR = 2017
  const CURR_YEAR = new Date().getFullYear()

  return (
    <footer className="footer text-center">
      <div className="footer__credits">
        <div className="container-fluid">
          <div className="last-child-mb-0">
            <p className="mb-0">
              Made with <a href="https://facebook.github.io/react/" target="_blank" className="footer__icon link-inherit" title="React"><Icon glyph="react" altText="React"/></a>{/*, <a href="http://sass-lang.com/" target="_blank" className="footer__icon link-inherit" title="SASS"><Icon glyph="sass" altText="SASS"/></a>*/}, <a href="https://webpack.js.org/" target="_blank" className="footer__icon link-inherit" title="Webpack"><Icon glyph="webpack" altText="Webpack 2"/></a>, <a href="https://www.contentful.com/" target="_blank" className="footer__icon link-inherit" title="Contentful"><Icon glyph="contentful" altText="Contentful"/></a> and <span className="footer__icon"><Icon glyph="heart" altText="love"/></span>
            </p>

            <p>
              &copy; {CURR_YEAR > CREATION_YEAR
                ? <span className="footer__credits__date">{CREATION_YEAR}&mdash;{CURR_YEAR}</span>
                : <span className="footer__credits__date">{CURR_YEAR}</span>
              } Edward Cobbold
            </p>
          </div>
        </div>
      </div>
    </footer>
  )
}

export default Footer
