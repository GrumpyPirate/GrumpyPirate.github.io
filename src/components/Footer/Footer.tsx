import Color from 'color';
import React, { FunctionComponent } from 'react';
import styled from 'styled-components';

import { Container } from 'components/Grid';
import Icon from 'components/Icon/Icon';
import REPO_URL from 'config/urls';
import { mediaQueries, palette, rem } from 'styles';
import { PropsWithClassName } from 'types/common';

import { Copyright, FooterLink, FooterLinkWithIcon, Tech } from './Footer.constants';

const Footer: FunctionComponent<PropsWithClassName> = function ({ className }) {
  const creationYear = 2017;
  const currentYear = new Date().getFullYear();

  return (
    <footer className={className}>
      <Container>
        <Tech>
          Made with{' '}
          <FooterLinkWithIcon
            href="https://facebook.github.io/react/"
            target="_blank"
            rel="noopener noreferrer"
            title="React"
          >
            <Icon glyph="react" altText="React" />
          </FooterLinkWithIcon>{' '}
          ,{' '}
          <FooterLinkWithIcon
            href="https://webpack.js.org/"
            target="_blank"
            rel="noopener noreferrer"
            title="Webpack"
          >
            <Icon glyph="webpack" altText="Webpack" />
          </FooterLinkWithIcon>{' '}
          and{' '}
          <FooterLinkWithIcon
            href="https://www.contentful.com/"
            target="_blank"
            rel="noopener noreferrer"
            title="Contentful"
          >
            <Icon glyph="contentful" altText="Contentful" />
          </FooterLinkWithIcon>
          .{' '}
          <FooterLink href={REPO_URL} target="_blank" rel="noreferrer noopener">
            Source code
          </FooterLink>
          .
        </Tech>

        <Copyright>
          &copy;{' '}
          <span>
            {currentYear > creationYear && <>{creationYear} &mdash; </>}
            {currentYear}
          </span>{' '}
          <FooterLink
            href="https://github.com/grumpypirate"
            target="_blank"
            rel="noopener noreferrer"
          >
            Stephanie Cobbold
          </FooterLink>
          .
        </Copyright>
      </Container>
    </footer>
  );
};

export default styled(Footer)`
  background-color: ${palette.themeDarkShades};
  color: ${Color(palette.themeLightShades).alpha(0.5).string()};
  font-size: ${rem(12)};
  padding: ${rem(24)} 0;
  text-align: center;

  @media ${mediaQueries.md} {
    padding: 2rem 0;
  }
`;
