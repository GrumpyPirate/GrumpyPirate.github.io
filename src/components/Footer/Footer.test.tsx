import { render } from '@testing-library/react';
import React from 'react';

import Footer from './Footer';

describe('Components', () => {
  describe('Footer', () => {
    it('should render as expected, without crashing', () => {
      const { container } = render(<Footer />);

      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
