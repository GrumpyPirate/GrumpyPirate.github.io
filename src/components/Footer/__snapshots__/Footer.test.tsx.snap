// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components Footer should render as expected, without crashing 1`] = `
.c1 {
  width: 100%;
  padding-right: 0.75rem;
  padding-left: 0.75rem;
}

.c5 {
  position: relative;
  bottom: 0.125em;
  display: inline-block;
  height: 1em;
  width: 1em;
  fill: currentColor;
  stroke: currentColor;
  stroke-width: 0;
  vertical-align: middle;
}

.c2 {
  margin-bottom: 0;
}

.c6 {
  margin-bottom: 0;
}

.c3 {
  color: inherit;
  font: inherit;
}

.c3,
.c3:hover,
.c3:focus,
.c3:active {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c3:hover,
.c3:focus {
  color: #dc8381;
}

.c4 {
  font-size: 1.75em;
  vertical-align: middle;
}

.c0 {
  background-color: #302445;
  color: rgba(242,241,243,0.5);
  font-size: 0.75rem;
  padding: 1.5rem 0;
  text-align: center;
}

@media (min-width:48em) {
  .c1 {
    padding-right: 0.9375rem;
    padding-left: 0.9375rem;
  }
}

@media (min-width:48em) {
  .c0 {
    padding: 2rem 0;
  }
}

<footer
  class="c0"
>
  <div
    class="c1"
  >
    <p
      class="c2"
    >
      Made with
       
      <a
        class="c3 c4"
        href="https://facebook.github.io/react/"
        rel="noopener noreferrer"
        target="_blank"
        title="React"
      >
        <svg
          aria-label="React"
          class="c5"
        >
          <use
            xlink:href="#react"
          />
        </svg>
      </a>
       
      ,
       
      <a
        class="c3 c4"
        href="https://webpack.js.org/"
        rel="noopener noreferrer"
        target="_blank"
        title="Webpack"
      >
        <svg
          aria-label="Webpack"
          class="c5"
        >
          <use
            xlink:href="#webpack"
          />
        </svg>
      </a>
       
      and
       
      <a
        class="c3 c4"
        href="https://www.contentful.com/"
        rel="noopener noreferrer"
        target="_blank"
        title="Contentful"
      >
        <svg
          aria-label="Contentful"
          class="c5"
        >
          <use
            xlink:href="#contentful"
          />
        </svg>
      </a>
      .
       
      <a
        class="c3"
        href="https://github.com/GrumpyPirate/GrumpyPirate.github.io"
        rel="noreferrer noopener"
        target="_blank"
      >
        Source code
      </a>
      .
    </p>
    <p
      class="c6"
    >
      ©
       
      <span>
        2017
         — 
        2020
      </span>
       
      <a
        class="c3"
        href="https://github.com/grumpypirate"
        rel="noopener noreferrer"
        target="_blank"
      >
        Edward Cobbold
      </a>
      .
    </p>
  </div>
</footer>
`;
